api/src/main/java/org/eclipse/microprofile/config/spi/Converter.java

```diff
@@ -114,13 +114,29 @@
  * All <em>built in</em> Converters have a priority value of {@code 1}. <em>Implicit</em> converters are only created
  * when no other converter was found; therefore, they do not have a priority.
  *
+ * <h3 id="empty">Empty values</h3>
+ *
+ * For all converters, the empty string {@code ""} <em>must</em> be considered an empty value. Some converters <em>may</em>
+ * consider other values to be empty as well.
+ * <p>
+ * When a conversion results in an empty value, most converters should normally return {@code null}.  All of the built-in
```

Comment:
I find this confusing an unnecessary. Does that mean even if source lookup ends up with `null` (not present) a converter could turn this into some value (present)? Effectively a programmed default value?

I suggest converters must not be burdened with handling `null` or empty/blank and `null` after source lookup should always result in the _not present_ result of the `Config` API. Should a user resolve a converter directly and pass an empty string or `null` to it it would be undefined behaviour. Or if we want to support this as a feature I feel the converter should behave consistent with the `Config` API, not with the `Source` API and throw an exception.

I understand that for some types there does exist a value that could represent an empty value or null but as it is impossible to formulate such a value using a source I don't see why converter would need to deal with them.

Reply:
> I find this confusing an unnecessary. Does that mean even if source lookup ends up with null (not present) a converter could turn this into some value (present)? Effectively a programmed default value?

I don't think so. The `org.eclipse.microprofile.config.spi.Converter#convert` states that callers should not pass `null` or you may get a `NullPointerException`.

Reply:
In which case I think for consistency empty/blank should be no different. And as it would be a pain to tax each converter to check for blank strings I suggest to assume they never get passed such as this is already an identified _not present_ result. If user still pass such values manual behaviour would be undefined.

Reply:
I am ok to remove this as it is implementation details based on what @jbee said.  @radcortez ok?

Reply:
Sure.